dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -84,12 +84,12 @@ class TeacherApplication < ApplicationBase
 
     has_many :emails, class_name: 'Pd::Application::Email', foreign_key: 'pd_application_id'
 
+    before_validation :set_course_from_program, unless: -> {program.nil?}
     validates :status, exclusion: {in: ['interview'], message: '%{value} is reserved for facilitator applications.'}
-    validates :course, presence: true, inclusion: {in: VALID_COURSES}, unless: -> {status == 'incomplete'}
```

Comment:
I moved this to be `before_save` because the controller test "can submit an empty form if application is incomplete" fails if this happens on `validates`. It appears that `set_status` will not get called in that test unless this line happens `before_save` (also `application.user.school_info` is nil if this line is in `validates`.)

I'm not exactly sure what's happening here. If someone has an idea about the why, or a strong reason why this line can't happen in `before_save`, let me know.